src/test/java/com/teragrep/pth_06/planner/walker/conditions/StreamingConditionTest.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.pth_06.planner.walker.conditions;
+
+import nl.jqno.equalsverifier.EqualsVerifier;
+import org.jooq.Condition;
+import org.jooq.impl.DSL;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StreamingConditionTest {
+
+    @Test
+    public void testValidTags() {
+        String[] tags = {
+                "index", "sourcetype", "host"
+        };
+        String value = "123456";
+        String operation = "EQUALS";
+        List<Condition> conditions = new ArrayList<>();
+        int loops = 0;
+        for (String tag : tags) {
+            Condition result = Assertions
+                    .assertDoesNotThrow(() -> new StreamingCondition(tag, value, operation).condition());
+            Assertions.assertNotEquals(DSL.noCondition(), result);
+            conditions.add(result);
+            loops++;
+        }
+        Assertions.assertEquals(3, loops);
+        List<Condition> expectedResults = new ArrayList<>();
```

Comment:
If you declare the list with `Arrays.asList(...)` it would be immutable. I would also change all expected variables and parameters to be `final`, just to make sure they are not changed at any moment.

Reply:
refactored to use `Arrays.asList` and `final` values in test cases in the scope of this PR.